### PR TITLE
fixed stay in lane when node.m_segment0 == 0

### LINKED ISSUE #618: StayInLane(StayInLaneMode.Forward/Backward) when node.m_segment0 == 0
The following code assumes that node.m_segment0 always exits:
(from https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition)
```c#
https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/blob/ab806f57362a2db7decffdbb336da5b64b5cdc74/TLM/TLM/UI/SubTools/LaneConnectorTool.cs#L372
if ((sourceLaneMarker.SegmentIndex == 0) ^ (stayInLaneMode == StayInLaneMode.Backward)) {
    continue;
}
```
Therefore when node.m_segment0 == 0 and the user tries to stay in lane only in one direction it will not work.

Reproduce:
1. create a segment
2. connect to more segments to it in a T shaped junction.
3. bulldoze the segment from step 1
4. use the lane connector tool to select the node and press SHIFT+S multiple times.

Suggested fix:
Find the smallest `i` where `node.GetSegment(i) ~= 0`
replace `sourceLaneMarker.SegmentIndex == 0` with `sourceLaneMarker.SegmentIndex == i`

### PR DESCRIPTION
**Pull request by kianzarrin**
fixes #618 please read the issue for details.

### CODE REVIEW

File: TLM/TLM/UI/SubTools/LaneConnectorTool.cs [hunk 1]
```diff
@@ -279,6 +279,19 @@ bool drawMarker
             return bounds.IntersectRay(mouseRay);
         }
 
+        /// <summary>
+        /// Finds the first index for which node.GetSegment(index) != 0 (its possible node.m_segment0 == 0)
+        /// </summary>
+        private static int GetFirstSegmentIDX(NetNode node) {
```
Comment (kvakvs) [inline TLM/TLM/UI/SubTools/LaneConnectorTool.cs]: Naming: Write `Index` not `IDX` unless its a some special abbreviation.

File: TLM/TLM/UI/SubTools/LaneConnectorTool.cs [hunk 2]
```diff
@@ -361,15 +374,15 @@ bool drawMarker
 
                         if (nodeMarkers != null) {
                             selectedMarker = null;
-
+                            int forwardSegmenIDX = GetFirstSegmentIDX(nodesBuffer[SelectedNodeId]);
```
Comment (kvakvs) [inline TLM/TLM/UI/SubTools/LaneConnectorTool.cs]: Naming: Typo in `segmen` and write `Index` unless its a some special abbreviation

**Comment by krzychu124** [comment]
@kianzarrin conflict 😕  
Fix and I will merge it 🙂

**Comment by kianzarrin** [comment]
@krzychu124 
> @kianzarrin conflict 😕  Fix and I will merge it 🙂
DONE!

**Pull request merged**